#!/usr/bin/env python
PACKAGE = "uav_commander"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("Kp_x", double_t, 0, "Proportional gain for x ", 0.075, 0, 2 )
gen.add("Kp_y", double_t, 0, "Proportional gain for y ", 0.075, 0, 2 )
gen.add("Kp_z", double_t, 0, "Proportional gain for z ", 0, 0, 2 )
gen.add("Kp_w", double_t, 0, "Proportional gain for w ", 0, 0, 2 )

gen.add("Ki_x", double_t, 0, "Integral gain for x ", 0.005, 0, 2 )
gen.add("Ki_y", double_t, 0, "Integral gain for y ", 0.005, 0, 2 )
gen.add("Ki_z", double_t, 0, "Integral gain for z ", 0, 0, 2 )
gen.add("Ki_w", double_t, 0, "Integral gain for w ", 0, 0, 2 )

gen.add("Kd_x", double_t, 0, "Differential gain for x ", 0.005, 0, 2 )
gen.add("Kd_y", double_t, 0, "Differential gain for y ", 0.005, 0, 2 )
gen.add("Kd_z", double_t, 0, "Differential gain for z ", 0, 0, 2 )
gen.add("Kd_w", double_t, 0, "Differential gain for w ", 0, 0, 2 )

gen.add("PID_Control", bool_t, 0, "Enable The PID Controller" , False)

exit(gen.generate(PACKAGE, "uav_commander", "PIDControl"))
